holoviews/core/spaces.py

```diff
@@ -752,10 +780,33 @@ def __getitem__(self, key):
         if self.call_mode == 'counter':
             val = val[1]
 
+        if data_slice:
+            val = self._dataslice(val, data_slice)
         self._cache(tuple_key, val)
         return val
 
 
+    def select(self, selection_specs=None, **kwargs):
+        selection = super(DynamicMap, self).select(selection_specs, **kwargs)
+        def dynamic_select(obj):
```

Comment:
Would ``dynamic_select`` have any use on its own in ``util``? It seems to offer a ``select`` that works with both dynamic and non dynamic objects..